functional-tests/src/test/java/com/redhat/che/functional/tests/BayesianPackageJson.java

```diff
@@ -0,0 +1,136 @@
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.pageobject.CodenvyEditor;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.Loader;
+import org.eclipse.che.selenium.pageobject.NavigateToFile;
+import org.eclipse.che.selenium.pageobject.NotificationsPopupPanel;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+public class BayesianPackageJson {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BayesianPackageJson.class);
+
+  @InjectTestWorkspace(template = RhCheWorkspaceTemplate.RH_NODEJS)
+  private TestWorkspace workspace;
+
+  @Inject private NavigateToFile navigateToFile;
+  @Inject private Loader loader;
+  @Inject private CodenvyEditor editor;
+  @Inject private Ide ide;
+  @Inject private NotificationsPopupPanel notificationsPopupPanel;
+  @Inject private ProjectExplorer projectExplorer;
+
+  private static final Integer JSON_EXPECTED_ERROR_LINE = 12;
+  private static final Integer JSON_INJECTION_ENTRY_POINT = 12;
+  private static final String PROJECT_FILE = "package.json";
+  private static final String PROJECT_NAME = "nodejs-hello-world";
+  private static final String PROJECT_DEPENDENCY = "\"serve-static\": \"1.7.1\" ,\n";
+  private static final String ERROR_MESSAGE =
+      "Package serve-static-1.7.1 is vulnerable: CVE-2015-1164. Recommendation: use version 1.7.2";
+
+  @BeforeClass
+  public void openTestFile() throws Exception {
+    try {
+      LOG.info(
+          "Workspace with name: "
+              + workspace.getName()
+              + " and id: "
+              + workspace.getId()
+              + " was successfully injected. ");
+      ide.open(workspace);
+      ide.waitOpenedWorkspaceIsReadyToUse();
+      projectExplorer.waitProjectExplorer();
+      notificationsPopupPanel.waitProgressPopupPanelClose();
+      projectExplorer.waitItem(PROJECT_NAME);
+    } catch (ExecutionException | InterruptedException e) {
+      LOG.error(
+          "Could not obtain workspace name and id - worskape was probably not successfully injected.");
+      throw e;
+    } catch (Exception e) {
+      LOG.error("Could not open workspace IDE.");
+      throw e;
+    }
+  }
+
+  @BeforeMethod
+  public void prepareProjectFile() {
+    openDefinedClass();
+    appendDependency();
+  }
+
+  @AfterMethod
+  public void closeFiles() {
+    removeDependency();
+    editor.closeAllTabs();
+  }
+
+  @Test
+  public void createBayesianError() {
+    try {
+      editor.waitTextInToolTipPopup(ERROR_MESSAGE);
+    } catch (Exception e) {
+      LOG.error(
+          "Bayesian error not present after adding dependency - known issue for prod-preview.");
```

Comment:
Would you like to show this error even if this fails on prod? I don't see any switch or check if the test runs on prod-preview. Same for ```checkErrorPresentAfterReopenFile``` method.

Reply:
fixed